autoPyTorch/utils/single_thread_client.py

```diff
@@ -0,0 +1,86 @@
+import typing
+from pathlib import Path
+
+import dask.distributed
+
+
+class DummyFuture(dask.distributed.Future):
+    """
+    A class that mimics a distributed Future, the outcome of
```

Comment:
Could you explain a bit more about why we need this? maybe in general what this whole file is doing?